diff --git a/configs/phycore-imx8mp_defconfig b/configs/phycore-imx8mp_defconfig
index b500a96d..a6e79415 100644
--- a/configs/phycore-imx8mp_defconfig
+++ b/configs/phycore-imx8mp_defconfig
@@ -66,6 +66,7 @@ CONFIG_CMD_EXT4=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_MTDPARTS=y
+CONFIG_CMD_FS_UUID=y
 CONFIG_OF_CONTROL=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
@@ -162,3 +163,4 @@ CONFIG_IMX_WATCHDOG=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_FDT_FIXUP_PARTITIONS=y
 CONFIG_LMB_MAX_REGIONS=10
+CONFIG_USE_BOOTCOMMAND=y
\ No newline at end of file
diff --git a/include/configs/phycore_imx8mp.h b/include/configs/phycore_imx8mp.h
index c49c2490..292dd0f2 100644
--- a/include/configs/phycore_imx8mp.h
+++ b/include/configs/phycore_imx8mp.h
@@ -156,15 +156,11 @@
 
 #define CONFIG_BOOTCOMMAND \
 	"mmc dev ${mmcdev}; if mmc rescan; then " \
-		"env exists dofitboot || setenv dofitboot 0;" \
-		"env exists doraucboot || setenv doraucboot 0;" \
-		"run spiprobe; " \
-		"if test ${doraucboot} = 1; then " \
-			"run raucboot; " \
-		"elif run loadimage; then " \
-			"run mmcboot; " \
-		"else run netboot; " \
-		"fi; " \
+		"if fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} boot.scr; then " \
+            "source ${loadaddr}; " \
+        "else " \
+            "echo 'cannot boot from: boot.scr'; " \
+        "fi; " \
 	"fi; " \
 	BOOTCOMMAND_APPEND
 
